### ISSUE TITLE
Strings chapter: &String doesn't actually coerce to &str

### ISSUE THREAD

**Issue by FreeFull** [rust-lang/rust#21850]
"Strings will coerece into &str with an &:
```
fn takes_slice(slice: &str) {
    println!("Got: {}", slice);
}

fn main() {
    let s = "Hello".to_string();
    takes_slice(&s);
}
```
"
This example should have `&*s` rather than `&s`, and the line above it should be reworded, since this isn't a coercion. (Note coerce is misspelled as coerece)

**Comment by retep998** [comment]
`&s` is actually valid now due to #21351.

**Comment by FreeFull** [comment]
Ah, I've missed that. The spelling mistake is the only bug, then.

**Comment by FreeFull** [comment]
https://github.com/rust-lang/rust/pull/21852 Fix for the spelling mistake.

**Comment by alexcrichton** [comment]
Fixed in #21852
